src/net_processing.cpp

```diff
@@ -699,7 +699,9 @@ void UpdateTxRequestTime(const uint256& txid, int64_t request_time) EXCLUSIVE_LO
 void RequestTx(CNodeState* state, const uint256& txid, int64_t nNow) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
 {
     CNodeState::TxDownloadState& peer_download_state = state->m_tx_download;
-    if (peer_download_state.m_tx_announced.size() >= MAX_PEER_TX_ANNOUNCEMENTS || peer_download_state.m_tx_announced.count(txid)) {
+    if (peer_download_state.m_tx_announced.size() >= MAX_PEER_TX_ANNOUNCEMENTS ||
+            peer_download_state.m_tx_process_time.size() >= MAX_PEER_TX_ANNOUNCEMENTS ||
```

Comment:
Maybe worth noting that I *think* `m_tx_process_time` is liable to outgrow `m_tx_announced` temporarily (e.g. we ask peer for txid, get back NOTFOUND, `m_tx_announced` entry immediately erased but `m_tx_process_time` entry hangs around until next `SendMessages()`) but I think this is negligible.